orienteer-core/src/main/java/org/orienteer/core/tasks/ITaskSession.java

```diff
@@ -1,9 +1,10 @@
 package org.orienteer.core.tasks;
 
 /**
- * Interface for task sessions 
+ * Interface for task sessions
+ * @param <T> task session type
  */
-public interface ITaskSession {
+public interface ITaskSession<T extends OTaskSession> {
```

Comment:
That's actually redundunt. You can always to specify more nearow result type in subclasses.

Reply:
Fixed in 37f0106